docs/DeveloperGuide.md

```diff
@@ -1,6 +1,24 @@
 # Developer Guide
 
+## Table of Contents 
```

Comment:
Thanks for adding this! I feel like there might be an automatic way of generating TOCs, but maybe not worth it to look into at this point.